File: parent/pinyougou-interface/src/main/java/cn/itcast/core/service/OrderService.java
package cn.itcast.core.service;

import cn.itcast.core.pojo.order.Order;
import entity.PageResult;
import pojogroup.OrderVoo;
import pojogroup.SalesVo;

import java.text.ParseException;
import java.util.Date;
import java.util.List;

public interface OrderService {
    void add (Order order);

    PageResult search (Integer page, Integer rows, Order order);

    int findCount ();

    Order findOne(Long id);

    List<OrderVoo> statisticsSales(Date beginTime, Date endTime);

    List<SalesVo> zhexian(Date beginTime, Date endTime) throws ParseException;

    void orderDeliver(Long[] ids);

    List<Order> selectAll ();
}

File: parent/pinyougou-web-manager/src/main/webapp/js/service/orderService.js
app.service('orderService', function ($http) {
        this.findAll = function () {
            return $http.get('../order/findAll.do');
        }
        //分页
        this.findPage = function (page, rows) {
            return $http.get('../order/findPage.do?page=' + page + '&rows=' + rows);
        }
        //查询实体
        this.findOne = function (id) {
            return $http.get('../order/findOne.do?id=' + id);
        }

        this.search = function (page, rows, searchEntity) {
            return $http.post('../order/search.do?page=' + page + "&rows=" + rows, searchEntity);
        }
        //查询订单数
        this.findCount = function () {
            return $http.get('../order/findCount.do');
        }

        this.importData = function () {
            return $http.get("../order/importData.do");
        }

    }
)